extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -726,18 +733,103 @@ export class AmpLightboxViewer extends AMP.BaseElement {
+        transLayer = this.element.ownerDocument.createElement('div');
+        transLayer.classList.add('i-amphtml-lightbox-viewer-trans');
+        this.element.ownerDocument.body.appendChild(transLayer);
 
-    anim.add(0, tr.setStyles(this.element, {
-      opacity: tr.numeric(1, 0),
-    }), 0.9, EXIT_CURVE_);
+        const rect = layoutRectFromDomRect(this.sourceElement_
+            ./*OK*/getBoundingClientRect());
+        const clone = image.cloneNode(true);
+        st.setStyles(clone, {
+          position: 'absolute',
+          top: st.px(imageBox.top),
+          left: st.px(imageBox.left),
+          width: st.px(imageBox.width),
+          height: st.px(imageBox.height),
+          transform: '',
+          transformOrigin: 'top left',
+          willChange: 'transform',
+        });
+        transLayer.appendChild(clone);
 
-    return anim.start(dur).thenAlways(() => {
-      this./*OK*/collapse();
-      st.setStyles(this.element, {opacity: ''});
+        // Fade out the container.
+        anim.add(0, tr.setStyles(dev().assertElement(this.container_), {
+          opacity: tr.numeric(1, 0),
+        }), 0.1, EXIT_CURVE_);
+
+        // Move and resize the image back to where it is in the article.
+        const dx = rect.left - imageBox.left;
+        const dy = rect.top - imageBox.top;
+        const scaleX = imageBox.width != 0 ? rect.width / imageBox.width : 1;
+        /** @const {!TransitionDef<void>} */
+        const moveAndScale = tr.setStyles(clone, {
+          transform: tr.concat([
+            tr.translate(tr.numeric(0, dx), tr.numeric(0, dy)),
+            tr.scale(tr.numeric(1, scaleX)),
+          ]),
+        });
+
+        // Duration will be somewhere between 0.2 and 0.8 depending on how far
+        // the image needs to move. Start the motion later too, but no later
+        // than 0.2.
+        const motionTime = Math.max(0.2,
+            Math.min(0.8, Math.abs(dy) / MAX_DISTANCE_APPROXIMATION * 0.8));
+        anim.add(Math.min(0.8 - motionTime, 0.2), (time, complete) => {
+          moveAndScale(time);
+          if (complete) {
+            this.sourceElement_.classList.remove('i-amphtml-ghost');
+          }
+        }, motionTime, EXIT_CURVE_);
+
+        // Fade out the transition image.
+        anim.add(0.8, tr.setStyles(transLayer, {
+          opacity: tr.numeric(1, 0.01),
+        }), 0.2, EXIT_CURVE_);
+
+        duration = this.getTransitionDuration_(dy);
+      }
+    }).then(() => {
+      return anim.start(duration).thenAlways(() => {
+        return this.vsync_.mutatePromise(() => {
+          if (this.sourceElement_) {
+            this.sourceElement_.classList.remove('i-amphtml-ghost');
+          }
+          st.setStyles(this.element, {
+            opacity: '',
+          });
+          st.setStyles(dev().assertElement(this.container_), {opacity: ''});
+          if (transLayer) {
+            this.element.ownerDocument.body.removeChild(transLayer);
```

Comment:
It would be bad if this does not happen, but the `if()` indicates `tansLayer` could be null. If we can gaurantee it won't be null let's remove the if. If we can't guarantee, we need to requery the node (maybe based on `i-amphtml-lightbox-viewer-trans` class) and remove it.